### PR TITLE
[ENH] add TimesNetModel v2 (BaseModel) with package layer

### PR DESCRIPTION
**Pull request by ShreyanshPagariya-13**
Adds `TimesNet` to v2

#### What should a reviewer concentrate their feedback on?

- **`_timesnet_layers.py`** — correctness of the `TimesBlock` period-padding
  logic and the `FFT_for_Period` amplitude weighting / aggregation step.
- **`_timesnet_v2.py` `forward()`** — the order of operations, though matches the original
  DSIPTS implementation.
- **`cat_cardinalities` validation** — whether the `ValueError` raised on
  mismatch between `len(cat_cardinalities)` and `metadata["encoder_cat"]` is
  the right approach, or whether silent padding / truncation would be preferable.
- **Output shape** — model returns `(B, H, n_targets)`, reviewers should confirm this is correct for the loss functions
  currently supported in v2.

#### Did you add any tests for the change?

No dedicated test file has been added yet. The implementation has been manually
verified.

#### Any other comments?

- This PR is part of the ongoing pytorch-forecasting v2 / sktime integration
  effort tracked in issue #1736.

#### PR checklist

- [x] The PR title starts with `[ENH]` — adding new model code.
- [ ] Added/modified tests — manual tests pass; formal test file to be added in a follow-up.
- [ ] Used pre-commit hooks when committing to ensure that code is compliant with hooks.
  Install hooks with `pre-commit install`.
  To run hooks independent of commit, execute `pre-commit run --all-files`.

### CODE REVIEW

File: pytorch_forecasting/layers/_timesnet/_timesnet_layers.py
Comment (phoeenniixx) [inline pytorch_forecasting/layers/_timesnet/_timesnet_layers.py:1]: please add this to `layers` module

File: pytorch_forecasting/models/timesnet/_timesnet_v2.py [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        """Return the package-layer class for this model."""
+        from pytorch_forecasting.models.timesnet._timesnet_v2_pkg import (
+            TimesNet_pkg_v2,
+        )
+
+        return TimesNet_pkg_v2
+
+    def __init__(
+        self,
+        metadata: dict,
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/timesnet/_timesnet_v2.py]: i think you should move it after the args with default values?

File: pytorch_forecasting/models/timesnet/_timesnet_v2_pkg.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        list of dict
+            Three parameter sets covering:
+
+            * minimal architecture (fast, default loss MAE)
+            * medium architecture with explicit MAE loss and custom lengths
+            * small architecture with MAPE loss and short sequences
+        """
+        from pytorch_forecasting.metrics import MAE, MAPE
+
+        params = [
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/timesnet/_timesnet_v2_pkg.py:49]: add the default param as well `{}`

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Thanks!
just added few comments

File: pytorch_forecasting/layers/_timesnet/_timesnet_layers.py
Comment (ShreyanshPagariya-13) [reply pytorch_forecasting/layers/_timesnet/_timesnet_layers.py:1]: Thanks for the review and suggestions, added the implementation to the `layers` module.

File: pytorch_forecasting/models/timesnet/_timesnet_v2.py (hunk 1 above)
Comment (ShreyanshPagariya-13) [reply pytorch_forecasting/models/timesnet/_timesnet_v2.py]: yaa following the convention i ordered them as needed.

File: pytorch_forecasting/models/timesnet/_timesnet_v2_pkg.py (hunk 2 above)
Comment (ShreyanshPagariya-13) [reply pytorch_forecasting/models/timesnet/_timesnet_v2_pkg.py:49]: Added the default parameter in the package configuration.
